test/Microsoft.AspNet.Mvc.ViewFeatures.Test/Rendering/HtmlHelperTextBoxTest.cs

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNet.Mvc.ModelBinding;
+using Microsoft.AspNet.Mvc.ModelBinding.Metadata;
+using Microsoft.AspNet.Mvc.TestCommon;
+using Microsoft.AspNet.Mvc.ViewFeatures;
+using Microsoft.AspNet.Testing;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq.Expressions;
+using Xunit;
+
+namespace Microsoft.AspNet.Mvc.Rendering
+{
+    public class HtmlHelperTextBoxTest
+    {
+        private const string InputWithPlaceholderAttribute = @"<input id=""HtmlEncode[[Property1]]"" name=""HtmlEncode[[Property1]]"" placeholder=""HtmlEncode[[placeholder]]"" type=""HtmlEncode[[{0}]]"" value="""" />";
+        private const string InputWithoutPlaceholderAttribute = @"<input id=""HtmlEncode[[Property1]]"" name=""HtmlEncode[[Property1]]"" type=""HtmlEncode[[{0}]]"" value="""" />";
+
+        [Theory]
+        [InlineData("text", InputWithPlaceholderAttribute)]
+        [InlineData("search", InputWithPlaceholderAttribute)]
+        [InlineData("url", InputWithPlaceholderAttribute)]
+        [InlineData("tel", InputWithPlaceholderAttribute)]
+        [InlineData("email", InputWithPlaceholderAttribute)]
+        [InlineData("number", InputWithPlaceholderAttribute)]
+        [InlineData("date", InputWithoutPlaceholderAttribute)]
+        [InlineData("color", InputWithoutPlaceholderAttribute)]
+        public void TextBoxFor_GeneratesPlaceholderAttribute_WhenDisplayAttributePromptIsSet(string type, string expectedResult)
```

Comment:
Agreed. Will do that and make sure the `InlineData` covers all of the relevant `type` attribute values for `input` between both theories.